### DISCUSSION TITLE
Big .sbc files

### DISCUSSION THREAD

**Discussion by johny65** [orientechnologies/orientdb#9494] [Q&A] [answered]
We have a database with 50+ millions of records that when compressed it occupies 4 GB, but we noted that uncompressed it is 120 GB of files, with lot of big `collections_xxxx_xxxxxxxxx.sbc` files, that are almost empty (lot of 0 bytes). We found [an old thread](https://www.mail-archive.com/orient-database@googlegroups.com/msg09449.html) with this information:

> *.sbc* files are used to store relations (edges) between vertexes. In sbc files space is occupied by removed edges reused only if it occupies half of total space to make possible to put all edges related to single vertex in close to each other on the disk.
Could you set parameter sbtreebonsai.freeeSpaceReuseTrigger to  0 and check result ?

Is there a way to "clean" this files in an existent database to reduce its size? We have no success with the mentioned parameter (`sbtreebonsai.freeSpaceReuseTrigger`). We also found in [this page](https://github.com/orientechnologies/orientdb-docs/blob/2.2.x/Console-Command-Show-Holes.md) the existence of `show holes` and `compact database` commands but only for version 2. What could we do in version 3?

**Comment by mdre** [comment]
Also, we export / Import database and we get a ~20GB used space after the process, so there are 100GB of waste space.

**Comment by johny65** [comment]
Some values:

- Original database in version 3.0: 120 GB.
- Exported and imported in 3.0: 20 GB.
- Exported and imported in 3.1: 15 GB.
- Exported and imported in 3.1 with SBTree Bonsai ridbags: 237 GB.

How is this possible?

**Comment by andrii0lomakin** [comment]
Hi guys, there is exactly the same issue in the bug tracker, we are working on it right now.

**Comment by andrii0lomakin** [comment]
Sorry, that is an issue created by the customer which uses commercial support. None the less we are working on this problem right now it will be released in 3.1.x and on ward version.

**Reply by johny65** [reply]
Yes, we are the commercial customer that created the issue 😅. We are now testing the snapshot build you send us. Thank you for your work.

**Reply by andrii0lomakin** [reply]
OK. :-) Please note that is not the final version, the final version should be faster, my primary concern now is testing of disk space consumption.

**Comment by johny65** [comment, marked as answer]
I don't see this documented anywhere, but if someone else is affected, version 3.2 solves this.
